src/ansys/motorcad/core/geometry.py

```diff
@@ -104,6 +104,28 @@ def remove_entity(self, entity_remove):
                     ):
                         self.entities.remove(entity)
 
+    def replace(self, replacement_region):
+        """Replace self with another region.
+
+        This method replaces region entities with entities from the replacement region object
+        (for example an imported DXF region).
+
+        Parameters
+        ----------
+        replacement_region : ansys.motorcad.core.geometry.Region
+            Motor-CAD region object, whose entities are to replace those of the
+            existing region.
+        """
+        # Remove existing entities from the region object
+        self.entities.clear()
+
+        # Get the list of replacement region entities
+        replacement_region_entities = replacement_region.entities
+
+        # Add the replacement region entities to the region object
+        for j in replacement_region_entities:
+            self.add_entity(j)
```

Comment:
Slight issue with this method. When you add the entities to this region it's only really adding a reference to the entity and not creating a new one. If the user deletes or edits the dxf region further in the script it will also edit the regions here. I think this is undesirable behaviour.

```Region.entities``` is derived from the Python list. See this for reference:
https://www.w3schools.com/python/python_lists_copy.asp

Reply:
@JackB-Ansys Let me know if you want a chat about this because object references/pointers might be new for you.